pyccel/parser/semantic.py

```diff
@@ -2131,27 +2131,33 @@ def _visit_Assign(self, expr, **settings):
             else:
 
                 # TODO check types from FunctionDef
-
                 master = macro.master
                 name = _get_name(master.name)
-
-                # all terms in lhs must be already declared and available
-                # the namespace
-                # TODO improve
+                results = []
+                args = [self._visit(i, **settings) for i in
+                            rhs.args]
+                args_names = [arg.name for arg in args if isinstance(arg, Variable)]
 
                 if not sympy_iterable(lhs):
                     lhs = [lhs]
+                results_shapes = macro.get_results_shapes(args)
+                for i, result in enumerate(macro.results):
+                    if result in macro.master_arguments and not lhs[i] in args_names:
```

Comment:
You use `lhs[i]` a several times in this loop. Why not use `zip` ? Either:
```
for i, (lhs_i, result) in enumerate(zip(lhs, macro.results)):
```
or:
```
for result, shape, lhs_i in zip(macro.results, results_shapes, lhs):
```

Reply:
Fixed here 0e3af3b